core/storage/base_model/gae_models.py

```diff
@@ -1067,7 +1067,7 @@ class BaseSnapshotMetadataModel(BaseModel):
     commit_type = ndb.StringProperty(
         required=True, choices=VersionedModel.COMMIT_TYPE_CHOICES)
     # The commit message associated with this snapshot.
-    commit_message = ndb.TextProperty(indexed=False)
+    commit_message = ndb.TextProperty(indexed=True)
```

Comment:
Change to ndb.StringProperty. See https://cloud.google.com/appengine/docs/standard/python/datastore/typesandpropertyclasses#StringProperty -- TextProperty cannot be indexed.

Could you also please update your job to yield:
- the number of strings with < 1200 bytes
- which strings are 1200-1499 bytes (provide enough info for these to identify the model itself -- the model type, ID, etc.)
- which strings are 1500+ bytes (provide enough info, as above). Also use a different top-level key from the second bullet point.

Rationale: we should take a closer look at these since https://cloud.google.com/appengine/docs/standard/python/datastore/typesandpropertyclasses#StringProperty says stringproperties can have at most 1500 bytes. The choice of 1200 above is to provide a bit of a buffer to see also if anything is close to the limit.

Double-check also that the backend handlers enforce a limit of 1000 bytes for commit messages, and ideally the frontend modals too.

Reply:
Hi Sean,

I think I have this working. One clarification question:

In this case do you want me to have a distinct key for each model with a property of length > 1200? This seems like it would be a lot of output...

Reply:
The key would be the same, but you would yield (key, values) at the end rather than (key, len(values)). The hope is that there aren't very many of these models. If you prefer, you can change 1200 to 1400.

Reply:
Got it. And to clarify further, I would still yield len(values) for the key for models with message size <1200, right?

Reply:
Yup.

Reply:
Done

Reply:
@varun-tandon Please read my original comment and make sure you have done every bit of it -- at least two parts are missing. https://github.com/oppia/oppia/pull/9963/files#r456994620

Reply:
Oops. Forgot to stage a change. Good catch. I see I missed the StringProperty, but what was the other thing I am missing. I addressed your changes to the keys. If you are referring to the backend limit, I checked and I observed a limit of 1500 characters (when saving an exploration).

Reply:
Yes, the backend limit -- do the handlers in the controller layer check for that and raise an error?

Reply:
I think the checks are only present on the frontend. I can find examples of places where the commit message field is not limited at the controller level. For example, in core.controllers.editor.py on line 121 (ExplorationHandler), you will see that the commit message is simply retrieved from the payload and the model is updated with the commit message without any length checks.

Reply:
Ah ok. Would it be possible to add validation at the controller level for all commit messages and enforce a limit? I think it can be lower (say 1000 chars) going forward.

Reply:
It's definitely possible. But I think we would also have to make a lot of frontend changes to update the limit for the users as well. Is this something you want me to start doing as a part of this PR?

Reply:
I think so, yeah -- since we're changing our backend limits, we need to make sure this is supported by the full stack.

Reply:
OK. Thankfully the frontend already handles issues with raising an InvalidInput exception, so the change was not too complicated.